### PR TITLE
Projects

### PR DESCRIPTION
**Pull request by Caleb335**
Hi @shahednasser

I've been able to make a simple layout.

This is how I approached this issue;
I created an array where I stored the details of every project that would be added to the website and mapped through it to display them in the DOM. For now the data there are dummy data and the real data for the UUID project.

Kindly take a look at the screenshot below and let me know what you think.
Screenshot from 2021-02-22 19-42-24

Issue: #1266

### CODE REVIEW

**Comment by Caleb335** [comment]
And I also think we could edit the contributing guideline so it'd accommodate how people can add different projects that they'd find in the future.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your work @Caleb335 I'm sorry for the delay as I have been busy. 

I like your approach I think it's great. I have just some small changes:

1. Let's change the title of Projects to Projects using sButtons
2. In the title of the page, change it to "Projects using sButtons - sButtons" (instead of sButtons being first then the title).
3.  Can you include a "How to add a project" section in the contribution guidelines as you have worked on it? It should be similar to the buttons examples part.

Once you do those changes remove the example projects and add the following projects that are using it please:
1. uuid (the one you have now)
2. [Grammarly with Ghost](https://github.com/shahednasser/grammarly-ghost)

Thank you so much for your work!

**Comment by Caleb335** [comment]
No problem @shahednasser .

I'd put all that you've said into consideration.

**Comment by Caleb335** [comment]
Hi @shahednasser, I've added the changes you requested. Kindly review, thanks.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,43 +15,46 @@ Please note we have a code of conduct, please follow it in all your interactions
...
-  * [Code of Conduct](#code-of-conduct)
-    + [Our Pledge](#our-pledge)
-    + [Our Standards](#our-standards)
-    + [Our Responsibilities](#our-responsibilities)
-    + [Scope](#scope)
-    + [Enforcement](#enforcement)
+  - [Notes on Creating New Buttons](#notes-on-creating-new-buttons)
+  - [Button Ideas](#button-ideas)
+  - [Create Pull Request](#create-pull-request)
+    - [Note on abandoned pull requests](#note-on-abandoned-pull-requests)
```
Comment (shahednasser) [inline CONTRIBUTING.md:21]: Add below this a list item linking to the section you added

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -111,22 +115,49 @@ If you would like to suggest a new button idea, please create a [new discussion]
...
+    about,
+    author,
+    link_to_project,
+    image,
+    github,
+  },
+```
+
+- When you are adding a new project, all you need to do is create another object that would contain the details of the project.
+- Make sure all the object keys are having the required string value.
```
Comment (shahednasser) [inline CONTRIBUTING.md:159]: Add that the person should add a screenshot or image for the project and where they should add it

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Caleb335) [reply CONTRIBUTING.md:21]: Alright, I'd work on that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Caleb335) [reply CONTRIBUTING.md:159]: Oops! I forgot  that.

I'd make the corrections.

**Comment by Caleb335** [comment]
Hi @shahednasser 

I've made the adjustments, kindly review, Thanks.

File: index.html [hunk 3]
```diff
@@ -46,45 +46,14 @@
     gtag("config", "UA-149400351-4");
   </script>
 </head>
-
+        <div class="links">
```
Comment (shahednasser) [inline index.html:49]: Why are these links before the body?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -15,43 +15,48 @@ Please note we have a code of conduct, please follow it in all your interactions
...
-    + [Our Standards](#our-standards)
-    + [Our Responsibilities](#our-responsibilities)
-    + [Scope](#scope)
-    + [Enforcement](#enforcement)
+  - [Notes on Creating New Buttons](#notes-on-creating-new-buttons)
+  - [Button Ideas](#button-ideas)
+  - [Create Pull Request](#create-pull-request)
+    - [Note on abandoned pull requests](#note-on-abandoned-pull-requests)
+    - [Adding Projects that make use of sButtons](#adding-projects-that-make-use-of-sbuttons)
+    - [Adding Projects that makes use sButtons](#adding-projects-that-make-use-of-sbuttons)
```
Comment (shahednasser) [inline CONTRIBUTING.md:23]: There are two links instead of one, please remove the second one

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: index.html (hunk 3 above)
Comment (Caleb335) [reply index.html:49]: I'm sorry about that.

I guess it's a mistake, I'd fix it now.

File: CONTRIBUTING.md (hunk 4 above)
Comment (Caleb335) [reply CONTRIBUTING.md:23]: On it

**Comment by Caleb335** [comment]
Kindly check now @shahednasser

**Comment by shahednasser** [comment]
Ok it's all great now but there's only one things required which is that in the latest change we added a dropdown menu for small devices. Please add it in the new pages well. You can copy it from index or any of the other old pages. You also need to add the new page in the dropdown menu

**Comment by Caleb335** [comment]
Hi @shahednasser I've made the changes you requested.

Kindly review.

File: projects.html [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+          <a href="index.html">
+            <img
+              src="assets/img/sbuttons-icon.png"
+              class="img-fluid"
+              width="64px"
+            />
+          </a>
+        </div>
+        <!-- dropdown menu on mobile -->
+        <div class="btn-group dropstart">
```
Comment (shahednasser) [inline projects.html:77]: Nest this under `.navbar-brand`

File: index.html [hunk 6]
```diff
@@ -1,240 +1,317 @@
...
-              <li class="nav-item">
-                <a class="nav-link" href="#">Link</a>
+              <li>
+                <a class="dropdown-item" href="examples.html">Examples</a>
               </li>
-              <li class="nav-item dropdown">
-                <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
-                  Dropdown
+              <li>
+                <a href="projects.html" class="link">
```
Comment (shahednasser) [inline index.html:105]: Change class to `dropdown-item` to match others in dropdown

**Review by shahednasser — CHANGES_REQUESTED** [review]
I have added a few fixes required. I also noticed that you didn't add the new link in the dropdown in the other pages but that's ok, we'll add it in another issue after merging this to give beginners a chance to work on it.

Also, I noticed that in the projects page in the cards, the links at the bottom in dark mode stay black. their text color should change to white.

File: projects.html (hunk 5 above)
Comment (Caleb335) [reply projects.html:77]: Ok, @shahednasser. Sorry about the late response

**Comment by Caleb335** [comment]
@shahednasser I've pushed the update to my branch.

The screenshot below shows how it looks in dark mode, Kindly take a look at it
Screenshot from 2021-03-06 16-04-03

**Comment by Caleb335** [comment]
But there's a little issue. You know the objects containing the attributes/details of a project are separated by a comma, so when it's (the projects) being rendered on the DOM the commas are appearing.

Dunno if you can look into that. I've tried figuring what the issue could be, all to no avail.

**Comment by shahednasser** [comment]
Regarding the commas, it's because you are using `map` which returns an array and then assigning it to `innerHTML`. You are basically assigning an array to innerHTML. Instead, you should create the elements and then append them using jQuery.

Another thing I noticed is that the drop down in Projects using sButtons is different than other pages so please fix that.

**Comment by Caleb335** [comment]
> Another thing I noticed is that the drop down in Projects using sButtons is different than other pages so please fix that.

On it

**Comment by Caleb335** [comment]
> Regarding the commas, it's because you are using map which returns an array and then assigning it to innerHTML. You are basically assigning an array to innerHTML. Instead, you should create the elements and then append them using jQuery.

Oh oh, I'd try fixing this too

**Comment by Caleb335** [comment]
@shahednasser Kindly take a look now. Thanks.

**Comment by shahednasser** [comment]
Amazing thank you for your hard work @Caleb335 !

**Pull request merged**